dragonfly/implementations/PsyllidProvider.py

```diff
@@ -0,0 +1,404 @@
+            self.request_status()
+        if self.status_value != 4:
+            logger.error('Could not activate Psyllid')
+            return False 
+        else: return True
+
+
+    def deactivate(self):
+        if self.status != 0:
+            logger.info('Deactivating Psyllid')
+            result = self.provider.cmd(self.psyllid_queue,'deactivate-daq')
+            self.request_status()
+        if self.status_value!=0:
+            logger.warning('Could not deactivate Psyllid')
+            return False
+        else: return True
+
+
+    def reactivate(self):
+        if self.status_value != 0:
+            logger.info('Reactivating Psyllid')
+            result = self.provider.cmd(self.psyllid_queue, 'reactivate-daq')
+            time.sleep(1)
+            self.request_status()
+            return True
+        elif self.status_value==0:
+	    self.activate()
+        else:
+             logger.warning('Cannot reactivate Psyllid')
+             return False
+
+
+    def quit_psyllid(self):
+        result = self.provider.cmd(self.psyllid_queue, 'quit-psyllid')
+        logger.info('psyllid quit')
+
+        
+    def get_all_central_frequencies(self):
+        return self.freq_dict
+    
+
+    def set_central_frequency(self, channel='a', cf=800e6):
+        cf_in_MHz = cf
+	
+        if self.mode == 'streaming':
+            request = '.active-config.ch'+str(self.channel_dict[channel])+'.strw.center-freq'
+            logger.info(request)
+            result = self.provider.set(self.psyllid_queue+request, cf_in_MHz)
+            logger.info('Set central frequency of streaming writer for channel {} to {} MHz'.format(channel, cf_in_MHz))
+            self.freq_dict[channel]=cf
+        elif self.mode == 'triggered':
+            request = '.active-config.ch'+str(self.channel_dict[channel])+'.trw.center-freq'
+            result = self.provider.set(self.psyllid_queue+request, cf_in_MHz)
+            logger.info('Set central frequency of egg writer for channel {} to {} MHz'.format(channel, cf_in_MHz))
+            self.freq_dict[channel]=cf
+
+
+    def set_all_central_frequencies(self, freq_dict):
+        for ichannel in freq_dict.keys():
+	    self.set_central_frequency(channel=ichannel, cf=freq_dict[ichannel])
+
+
+    def start_run(self, duration, filename):
+        if duration ==0:
+            return False
+        payload = {'duration':duration, 'filename':filename}
+        result = self.provider.cmd(self.psyllid_queue, 'start-run', payload=payload)
+        return True
+
+    
+    def get_number_of_streams(self):
+        active_channels = [i for i in self.freq_dict.keys() if self.freq_dict[i]!=None]
+        logger.info('Active channels are {}'.format(active_channels))
+        return len(active_channels)
+
+        
+    def get_active_streams(self):
+        active_channels = [i for i in self.freq_dict.keys() if self.freq_dict[i]!=None]
+        return active_channels
+
```

Comment:
just to nitpick, it should probably be 
```python
@property
def <whatever_thing>(self):
```
to avoid having to `dragonfly get psyllid_provider_name.get_<whatever_thing>` with extra get